### PR TITLE
upgrade: workaround potential network issues during the upgrade

### PR DESCRIPTION
**Pull request by rhafer**
After evacuating routers and stopping the remaining services reset the
fail-mode of the ovs bridges to keep network connectivty during the
package upgrade.
https://bugzilla.suse.com/show_bug.cgi?id=1002734

Additionally adds a workaround for https://bugzilla.suse.com/show_bug.cgi?id=1011889 by forcing a wicked ifreload call in the preupgrade script.

### CODE REVIEW

File: chef/cookbooks/crowbar/templates/default/crowbar-pre-upgrade.sh.erb [hunk 1]
```diff
@@ -26,3 +26,8 @@ service pacemaker stop
 echo "No HA setup found..."
 
 <% end %>
+
+for bridge in $(ovs-vsctl --real list-br); do
```
Comment (rhafer) [inline chef/cookbooks/crowbar/templates/default/crowbar-pre-upgrade.sh.erb]: It should be enough (an probably safer) if we just do this for bridges which have ports connected to external devices (usually br-public and br-fixed). And only if the ml2 ovs driver is used. Probably easiest to figure out the bridge names in crowbar and set them template variables.

File: chef/cookbooks/crowbar/templates/default/crowbar-pre-upgrade.sh.erb [hunk 2]
```diff
@@ -26,3 +26,8 @@ service pacemaker stop
...
+    echo "Resetting fail-mode on ovs-bridge $bridge"
+    ovs-vsctl del-fail-mode $bridge
```
Comment (abel-navarro) [inline chef/cookbooks/crowbar/templates/default/crowbar-pre-upgrade.sh.erb]: maybe we want to explicitly set the fail mode to *standalone* (`ovs-vsctl set-fail-mode bridge standalone`). This way openvswitchd can get over the missing controller (neutron agent).

**Comment by rhafer** [comment]
> I think we still need to investigate whether upgrading OVS disconnects the admin network so we loose connectivity to the node. Otherwise these changes may not be necessary at all.

I checked that already. I does disconnect the admin network, when using the vlan driver and nova_fixed is same interface as admin.

**Comment by jsuchome** [comment]
[networking after upgrade (bsc#1002734)](https://trello.com/c/07RTRTXu/178-networking-after-upgrade-bsc-1002734)

**Pull request merged**